### PR TITLE
[ALLUXIO-1819] Used diamond operator instead of Guava's shorthand

### PR DESCRIPTION
**Pull request by HorizonNet**
https://alluxio.atlassian.net/browse/ALLUXIO-1819

### CODE REVIEW

File: core/server/src/test/java/alluxio/worker/block/BlockMetadataManagerViewTest.java [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import org.mockito.Mockito;
 
 import java.io.File;
+import java.util.HashSet;
```
Comment (aaudiber) [inline core/server/src/test/java/alluxio/worker/block/BlockMetadataManagerViewTest.java]: lets remove all uses of Sets.new* so we can drop the Sets import

File: tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java [hunk 2]
```diff
@@ -48,6 +48,7 @@
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileReader;
+import java.util.ArrayList;
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java]: lets remove all uses of Lists.new* so we can drop the Lists import

**Comment by aaudiber** [comment]
@HorizonNet Thanks for this PR, I left a couple comments

**Comment by HorizonNet** [comment]
@aaudiber Thanks for the comments. I updated the PR accordingly. Please have a look at it. It now essentially does what Guava does under the hood. If you know a shorter way, just let me know.

**Comment by aaudiber** [comment]
@HorizonNet hm, it's unfortunate that new `ArrayList<>()` doesn't support arguments in the same way - could we use `Arrays.asList(...)` instead?

**Comment by aaudiber** [comment]
What are your thoughts @calvinjia?

File: core/server/src/test/java/alluxio/worker/block/BlockLockManagerTest.java [hunk 3]
```diff
@@ -261,11 +261,11 @@ public void stressTest() throws Throwable {
     final int lockUnlocksPerThread = 50;
     setMaxLocks(numBlocks);
     final BlockLockManager manager = new BlockLockManager();
-    final List<Thread> threads = Lists.newArrayList();
+    final List<Thread> threads = new ArrayList<>();
     final CyclicBarrier barrier = new CyclicBarrier(numBlocks * threadsPerBlock);
     // If there are exceptions, we will store them here.
     final AtomicReference<List<Throwable>> failedThreadThrowables =
-        new AtomicReference<List<Throwable>>(Lists.<Throwable>newArrayList());
+        new AtomicReference<List<Throwable>>(new ArrayList<Throwable>());
```
Comment (jsimsa) [inline core/server/src/test/java/alluxio/worker/block/BlockLockManagerTest.java]: could this be simplified to `new AtomicReference<>(new ArrayList<Throwable>())` or perhaps even `new AtomicReference<>(new ArrayList<>())`?

File: tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java [hunk 4]
```diff
@@ -211,21 +214,23 @@ public void docExampleTest() throws Exception {
     // input file paths
     AlluxioURI input1 = new AlluxioURI("/inputFile1");
     AlluxioURI input2 = new AlluxioURI("/inputFile2");
-    List<AlluxioURI> inputFiles = Lists.newArrayList(input1, input2);
+    ArrayList<AlluxioURI> files = new ArrayList<>();
```
Comment (jsimsa) [inline tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java]: please keep the name as `inputFiles`

File: tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java [hunk 5]
```diff
@@ -211,21 +214,23 @@ public void docExampleTest() throws Exception {
...
+    Collections.addAll(files, input1, input2);
     // output file paths
     AlluxioURI output = new AlluxioURI("/outputFile");
-    List<AlluxioURI> outputFiles = Lists.newArrayList(output);
+    ArrayList<AlluxioURI> files2 = new ArrayList<>();
```
Comment (jsimsa) [inline tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java]: pease keep the name as `outputFiles`

**Comment by jsimsa** [comment]
@HorizonNet thanks for the PR, did a pass and left some comments

File: core/server/src/test/java/alluxio/worker/block/BlockLockManagerTest.java (hunk 3 above)
Comment (HorizonNet) [inline core/server/src/test/java/alluxio/worker/block/BlockLockManagerTest.java]: Unfortunately, this doesn't seem to work.

**Comment by HorizonNet** [comment]
@jsimsa Thanks for the comments. I updated the PR accordingly.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**